### DISCUSSION TITLE
Merge two tRPC routers or at least mount both in Next.js when they have completely different contexts.

### DISCUSSION THREAD

**Discussion by ethndotsh** [trpc/trpc#5111] [Q&A]
I have two routers, one called AuroraRouter which is a reference to a Fastify tRPC server living in another folder in my monorepo, and AppRouter which is the local tRPC server in the Next.js app where I am trying to mount both. They both have very different contexts, and different auth requirements but both need to be used with React Query. What's the best way to mount/merge/use both using the pages dir, with tanstack query?

**Comment by Nick-Lucas** [comment]
You might benefit from this example: https://github.com/trpc/trpc/tree/main/examples/soa

**Comment by ethndotsh** [comment]
The issue with that one is that it seems that the contexts need to be the
same

**Reply by Nick-Lucas** [reply]
Right yes, well the context has to be provided at the adapter, so they probably do need to at least be structurally compatible.

you might be able to do something to merge them in the types but it’s not something we’ve ever had to do, so will be interesting if you can figure it out

**Reply by ethndotsh** [reply]
Alright - that’s a little difficult because the scope of the two tRPC servers is completely different.

**Reply by KATT** [reply]
They only need to be the same for the error formatting to work. So really that's all that should be needed technically, feel free to open a feature request to make this work

**Reply by ethndotsh** [reply]
Will do, thank you!

**Reply by ashburnham** [reply]
Did you manage it?

**Reply by ethndotsh** [reply]
No but I opened a feature request #5113

**Comment by rjay98** [comment]
FYI just adding some more context (variations of this have been floating around). I suspect a common point in the lifecycle of customers will involve a separate server migration with the ability to do piece-wise changes. Having global context definitely gets in the way of this. We're monkey-patching this a bit, but I can also foresee a future where different teams own different trpc servers that manage different services.
